From 060cb8f861564afafd8e6672ebe88f93dd4473dd Mon Sep 17 00:00:00 2001
From: Shreyas NC <shreyas.nc@intel.com>
Date: Wed, 8 Mar 2017 03:08:18 +0530
Subject: [PATCH 0562/1132] ASoC: Intel: Skylake: Define tokens for aggregation

To support aggregation, define tokens for aggregation id,
masters participating in aggregation, link id and channel mask.

Change-Id: Ib7e3f5a3aec4d8a6e2dec1b1f045c8078a3ea958
Signed-off-by: Shreyas NC <shreyas.nc@intel.com>
Reviewed-on:
Reviewed-by: Singh, Guneshwor O <guneshwor.o.singh@intel.com>
Reviewed-by: Jayanti, Satya Charitardha <satya.charitardha.jayanti@intel.com>
Reviewed-by: Prodduvaka, Leoni
Reviewed-by: Prusty, Subhransu S <subhransu.s.prusty@intel.com>
Reviewed-by: R, Dharageswari <dharageswari.r@intel.com>
Reviewed-by: Koul, Vinod <vinod.koul@intel.com>
Reviewed-by: Diwakar, Praveen <praveen.diwakar@intel.com>
Reviewed-by: Kale, Sanyog R <sanyog.r.kale@intel.com>
Tested-by: Avati, Santosh Kumar <santosh.kumar.avati@intel.com>
---
 include/uapi/sound/snd_sst_tokens.h | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/include/uapi/sound/snd_sst_tokens.h b/include/uapi/sound/snd_sst_tokens.h
index 8ba0112..5d3d81a 100644
--- a/include/uapi/sound/snd_sst_tokens.h
+++ b/include/uapi/sound/snd_sst_tokens.h
@@ -232,6 +232,16 @@
  *                              from source specified by clock source.
  *
  * %SKL_TKN_U32_ASTATE_CLK_SRC: Clock source for A-State entry
+ * %SKL_TKN_U32_AGG_NUM_MASTERS:
+ *                              Number of aggregated masters
+ *
+ * %SKL_TKN_U32_AGG_LINK_ID:    Aggregated master's instance id
+ *
+ * %SKL_TKN_U32_AGG_CH_MASK:    Represents channels driven by the master
+ *
+ * %SKL_TKN_U32_AGG_ID:         Aggregation id is a non zero identifier to
+ *                              indicate if this endpoint is participating
+ *                              in aggregation.
  *
  * module_id and loadable flags dont have tokens as these values will be
  * read from the DSP FW manifest
@@ -324,7 +334,13 @@ enum SKL_TKNS {
 	SKL_TKN_U32_ASTATE_COUNT,
 	SKL_TKN_U32_ASTATE_KCPS,
 	SKL_TKN_U32_ASTATE_CLK_SRC,
-	SKL_TKN_MAX = SKL_TKN_U32_ASTATE_CLK_SRC,
+
+	SKL_TKN_U32_AGG_NUM_MASTERS,
+	SKL_TKN_U32_AGG_LINK_ID,
+	SKL_TKN_U32_AGG_CH_MASK,
+	SKL_TKN_U32_AGG_ID,
+
+	SKL_TKN_MAX = SKL_TKN_U32_AGG_ID,
 };
 
 #endif
-- 
2.7.4

